crates/sui-core/src/authority.rs

```diff
@@ -1679,7 +1678,8 @@ impl AuthorityState {
         let name_type = move_object.type_().try_extract_field_name(&type_)?;
 
         let bcs_name = bcs::to_bytes(&name_value.clone().undecorate()).map_err(|e| {
-            SuiError::ObjectSerializationError {
+            SuiError::SerializationError {
+                format: "bcs_name".to_string(),
```

Comment:
maybe this is exposing too much?

Reply:
Alternatively this can just directly be constructed from `bcs::Error`, which is what it's mostly used for anyways

Reply:
Nvm, `bcs::Error` doesn't implement `Hash`, which is needed on `SuiError` (I think because this gets stored in authority-something somehow)